### PR TITLE
Decorate class

### PR DESCRIPTION
**Pull request by peterboro**
# Dear Reviewer(s)

## Thank you for taking the time to review this pull request.

I implemented the following: 
1. Create a class Nameable.
- Implement a method called correct_name that will raise a NotImplementedError.
2. Turn your Person class to Nameable
- Make sure that your Person class inherits from Nameable
- Make sure that this class has a method correct_name implemented. It should simply return the name attribute.
3. Prepare base Decorator
- Make sure that it inherits from Nameable.
- In the constructor assign a nameable object from params to an instance variable.
- Implement the correct_name method that returns the result of the correct_name method of the @nameable.
4 Prepare CapitalizeDecorator and TrimmerDecorator
For the CapitalizeDecorator:
- Create a class that inherits from the base Decorator class.
- Implement a method correct_name that capitalizes the output of @nameable.correct_name.
For the TrimmerDecorator:
- Create a class that inherits from the base Decorator class.
- Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
```
Comment (FrederickMih) [inline nameable.rb]: - [ ] Let's kindly raise the error this way:
`raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"`
[Pattern description](https://refactoring.guru/design-patterns/decorator/ruby/example#example-0)

File: decorator.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
```
Comment (FrederickMih) [inline decorator.rb:2]: - [ ] We need a constructor for nameable ie, `attr_accessor :nameable` Let's kindly fix :wrench: it

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @peterboro  , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


Screenshot from 2022-10-20 09-38-51


## Highlights :green_circle: 


- Correct Gitflow has been followed :heavy_check_mark: 
- Descriptive README.md file :heavy_check_mark: 
- PR has a good title and summary :heavy_check_mark: 
- No Linter error :heavy_check_mark: 


## Required Changes ♻️

- Let's kindly resolve the N+1 query issues when fetching all post and their comments. Let's terminate the issue in order to keep our application scalable. In Rails, you can use includes method for eager loading.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @peterboro,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**